packages/ag-charts-enterprise/src/features/annotations/annotations.ts

```diff
@@ -1162,13 +1177,9 @@ export class Annotations extends _ModuleSupport.BaseModuleInstance implements _M
         const shiftKey = (event.sourceEvent as MouseEvent).shiftKey;
 
         const offset = Vec2.from(event);
-        const point = (origin?: Point, angleStep: number = 1) =>
-            shiftKey
-                ? invertCoords(
-                      snapToAngle(offset, origin ? convertPoint(origin, context) : Vec2.origin(), angleStep),
-                      context
-                  )
-                : invertCoords(offset, context);
+        this.hoverCoords = offset;
+
+        const point = this.getPointFn(shiftKey, offset, context);
 
         state.transition('hover', { offset, point });
```

Comment:
The state machine could store the `offset` passed through here and then it wouldn't need to use `getHoverCoords()` and we could avoid storing that state on the `Annotations` module class.

Then the state machine would also have all the data that is passed to `getPointFn()`. So it should be able to do that internally as well. 

It has `snapping` state, the hover `offset` and the `context`, so we should be able to push all that down into the state machine.

We should try to avoid adding any state to the `Annotations` module class beyond the user options and `annotationData`.

Reply:
Discussed offline and decided to leave as is